### DISCUSSION TITLE
openapi is undefined

### DISCUSSION THREAD

**Discussion by arthurvanl** [samchungy/zod-openapi#344] [Q&A]
![afbeelding](https://github.com/user-attachments/assets/48cd15c7-9aa4-426d-b015-9f4d7dd2472b)
I am using it in bun

**Comment by samchungy** [comment]
Hey there I need a little bit more of a reproduction.

My assumption initially is maybe bun is stripping side effects?

**Comment by arthurvanl** [comment]
I tried recreating but did not really work out. I had it fixed for the tests somehow. Now I deployed a package using this tool (private tho) and got the error again.
![afbeelding](https://github.com/user-attachments/assets/a2067304-5084-4294-ac9b-70feea041fe7)

I have no clue about how I can resolve this.

**Comment by arthurvanl** [comment]
I just released a public version of the tool so that you can try it yourself
https://npmjs.com/package/openapi-builder-test

**Comment by arthurvanl** [comment]
![afbeelding](https://github.com/user-attachments/assets/011918eb-c019-4d7e-ace1-e564817d8465)
This is my file structure

schemas made in the schemas folder and the route paths I add in the api folder.

In the src/utils/redoc.ts is the specification variable created. which I then use in the src/index.ts


In src/index.ts I import the route paths using dynamic import. This will add the paths and the route + schemas.
Except it is never added because it says `openapi is undefined`

**Comment by arthurvanl** [comment]
I actually got a small reproduction using the test package:

use 
```sh
bun init
```

Create a folder inside this folder: `api` and create a `index.ts` file.
Add the following code to this file:
```ts
import { Parameter, PathBuilder, PathOperation, RequestBodyBuilder, ResponseBuilder, ResponseOptions } from "openapi-builder-test";
import { object, z } from "zod";

enum SessionStatus {
    Validated = 'validated',
    ValidateTwoFactor = 'validate_twofactor',
    Expired = 'expired'
}

export default new PathBuilder({
    path: '/users',
    operations: [{
        parameters: [] as Parameter[],
        requestBody: new RequestBodyBuilder()
        .setSchema('LoginSchema', z.object({
            message: z.literal('Credentials are correct!'),
            session: z.object({
                token: z.string(),
                status: z.nativeEnum(SessionStatus),
                user_id: z.string().uuid()
            })
        })).getProps(),
        responses: [new ResponseBuilder({schema: z.object({test: z.string()}).openapi({example: {test: 'asasdasd'}}), schemaName: 'asdasdasds'} as unknown as ResponseOptions).getProps()]
    } as PathOperation]
})
```

For the main file use this code:

```ts
import { OpenAPIBuilder, PathBuilder } from "openapi-builder-test";
import { dirname } from "node:path";
import { fileURLToPath } from "node:url";
import { readdir, readdirSync } from "node:fs"
import { write } from "bun"
import { z } from "zod";

const current = dirname(fileURLToPath(import.meta.url));

export const specification = new OpenAPIBuilder()
.setVersion('3.1.0')
.setPaths((path) => 
    path.setPath('/test')
    .setOperations((op) => 
        //! if you uncomment this and import zod, the error is gone.
        op.setRequestBody((rb) => 
            rb.setSchema('TestSchema', z.object({
                message: z.literal('testing this feature'),
                data: z.array(z.object({test: z.string()}))
            })
            // rb.setSchema('asd', true)
        ))
        .addResponse((r) => r.setSchema('TestSchema', z.object({message: z.literal('Success')}))))
        //! comment this and then it works
        // op.setRequestBody((rb) => rb.setSchema('asd', true))
        // .addResponse((r) => r.setSchema('TestSchema', true)))
    )


const files = readdirSync(current + '/api');

for(let i=0; i<files.length; i++) {
    const filePath = files[i];
    
    const test: {default: PathBuilder} = await import(`${current}/api/${filePath}`);
    specification.addPath(test.default)
}

write('test.json', JSON.stringify(specification.toJSON()));
```


If you test this. This would work. When zod import is removed and no zod is used in the main file but used in the other file it'll throw the error.

**Comment by samchungy** [comment]
So in your first file I don't see zod being extended anywhere?

**Comment by samchungy** [comment]
You have to call the function before you use .openapi

**Comment by arthurvanl** [comment]
it is extended in that package?

**Comment by samchungy** [comment]
I don't see it being extended anywhere

**Comment by arthurvanl** [comment]
![afbeelding](https://github.com/user-attachments/assets/314e76af-e7ef-45b4-a7e3-e994fef79464)

**Comment by arthurvanl** [comment]
file: src/utils/index.ts of openapi-builder-test

**Comment by samchungy** [comment]
That's within a function so it isn't getting called

**Comment by samchungy** [comment]
You can't use declare a variable with .openapi until after you've called the extend function.

**Comment by arthurvanl** [comment]
![afbeelding](https://github.com/user-attachments/assets/472f51c2-c537-4213-a690-fda9f0772e53)
so why does this not work then?

**Comment by arthurvanl** [comment]
I even added the extend function in the api/index.ts file, seems not take effect at all

**Comment by samchungy** [comment]
Can you try running the example in the readme. If this doesn't work, bun is stripping side effects and if you don't know what side effects are you want to look that up 

import { z } from 'zod';
import { extendZodWithOpenApi } from 'zod-openapi';

extendZodWithOpenApi(z);

z.string().openapi({ description: 'hello world!', example: 'hello world' });

**Comment by samchungy** [comment]
v5 now removes the extension
